pkg/scheduler/queue/simple.go

```diff
@@ -0,0 +1,133 @@
+	name:        "simplePlacementSchedulingQueue",
+}
+
+// Run starts the scheduling queue.
+//
+// At this moment, Run is an no-op as there is only one queue present; in the future,
+// when more queues are added, Run would start goroutines that move items between queues as
+// appropriate.
+func (sq *simplePlacementSchedulingQueue) Run() {}
+
+// Close shuts down the scheduling queue immediately.
+func (sq *simplePlacementSchedulingQueue) Close() {
+	sq.active.ShutDown()
+}
+
+// CloseWithDrain shuts down the scheduling queue and returns until all items are processed.
+func (sq *simplePlacementSchedulingQueue) CloseWithDrain() {
+	sq.active.ShutDownWithDrain()
+}
+
+// NextPlacementKey returns the next PlacementKey (either clusterResourcePlacementKey or resourcePlacementKey)
+// in the work queue for the scheduler to process.
+//
+// Note that for now the queue simply wraps a work queue, and consider its state (whether it
+// is shut down or not) as its own closedness. In the future, when more queues are added, the
+// queue implementation must manage its own state.
+func (sq *simplePlacementSchedulingQueue) NextPlacementKey() (key PlacementKey, closed bool) {
+	// This will block on a condition variable if the queue is empty.
+	placementKey, shutdown := sq.active.Get()
+	if shutdown {
+		return "", true
+	}
+	return placementKey.(PlacementKey), false
+}
+
+// Done marks a PlacementKey as done.
+func (sq *simplePlacementSchedulingQueue) Done(placementKey PlacementKey) {
+	sq.active.Done(placementKey)
+}
+
+// Add adds a PlacementKey to the work queue.
+//
+// Note that this bypasses the rate limiter (if any).
+func (sq *simplePlacementSchedulingQueue) Add(placementKey PlacementKey) {
+	sq.active.Add(placementKey)
+}
+
+// AddRateLimited adds a PlacementKey to the work queue after the rate limiter (if any)
+// says that it is OK.
+func (sq *simplePlacementSchedulingQueue) AddRateLimited(placementKey PlacementKey) {
+	sq.active.AddRateLimited(placementKey)
+}
+
+// AddAfter adds a PlacementKey to the work queue after a set duration.
+//
+// Note that this bypasses the rate limiter (if any).
+func (sq *simplePlacementSchedulingQueue) AddAfter(placementKey PlacementKey, duration time.Duration) {
+	sq.active.AddAfter(placementKey, duration)
+}
+
+// AddBatched tracks a PlacementKey and adds such keys in batch later to the work queue when appropriate.
+//
+// For the simple queue implementation, this is equivalent to Add.
+func (sq *simplePlacementSchedulingQueue) AddBatched(placementKey PlacementKey) {
+	sq.active.Add(placementKey)
+}
+
+// Forget untracks a PlacementKey from rate limiter(s) (if any) set up with the queue.
+func (sq *simplePlacementSchedulingQueue) Forget(placementKey PlacementKey) {
+	sq.active.Forget(placementKey)
+}
+
+// NewSimplePlacementSchedulingQueue returns a simplePlacementSchedulingQueue.
+func NewSimplePlacementSchedulingQueue(name string, rateLimiter workqueue.TypedRateLimiter[any]) PlacementSchedulingQueue {
```

Comment:
why change this function signature? AFAIK, this is not part of the new common interface